From ad36c41aac1953723a5614063c53b166d07345bf Mon Sep 17 00:00:00 2001
From: Zane van Iperen <zane@zanevaniperen.com>
Date: Fri, 17 Apr 2026 21:09:23 +1000
Subject: [PATCH 2/2] cl_dll/in_camera: allow thirdperson in multiplayer

---
 cl_dll/in_camera.cpp | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/cl_dll/in_camera.cpp b/cl_dll/in_camera.cpp
index 143b8c5f..45b61e4a 100644
--- a/cl_dll/in_camera.cpp
+++ b/cl_dll/in_camera.cpp
@@ -169,9 +169,6 @@ void DLLEXPORT CAM_Think( void )
 #endif
 	vec3_t viewangles;
 
-	if( gEngfuncs.GetMaxClients() > 1 && CL_IsThirdPerson() )
-		CAM_ToFirstPerson();
-
 	switch( (int)cam_command->value )
 	{
 		case CAM_COMMAND_TOTHIRDPERSON:
@@ -478,13 +475,6 @@ void CAM_OutUp( void )
 void CAM_ToThirdPerson( void )
 {
 	vec3_t viewangles;
-#if !_DEBUG
-	if( gEngfuncs.GetMaxClients() > 1 )
-	{
-		// no thirdperson in multiplayer.
-		return;
-	}
-#endif
 	gEngfuncs.GetViewAngles( (float *)viewangles );
 
 	if( !cam_thirdperson )
-- 
2.53.0

