diff --git a/nimrun.cpp b/nimrun.cpp
index 4be116a..c0d49bc 100644
--- a/nimrun.cpp
+++ b/nimrun.cpp
@@ -28,6 +28,7 @@
 #include <iomanip>
 #include <fstream>
 #include <iostream>
+#include <optional>
 #include "config.h"
 #include "nimrun.hpp"
 
diff --git a/nimrun.hpp b/nimrun.hpp
index 1942e10..cbfce26 100644
--- a/nimrun.hpp
+++ b/nimrun.hpp
@@ -23,6 +23,7 @@
 #include <memory>
 #include <unordered_map>
 #include <algorithm>
+#include <vector>
 #include <cstdio>
 #include <unistd.h>
 #include <dlfcn.h>
