diff --git a/agent/ssl.cpp b/agent/ssl.cpp
index 9942cbb..7c01385 100644
--- a/agent/ssl.cpp
+++ b/agent/ssl.cpp
@@ -117,9 +117,6 @@ x509_store_ptr nimrod::load_ca_store_mem(const char *data, size_t size)
 void nimrod::set_ssl_store(SSL_CTX *ctx, X509_STORE *st) noexcept
 {
 	SSL_CTX_set_cert_store(ctx, st);
-
-	/* SSL_CTX_set_cert_store() doesn't add a reference, so do it here. */
-	CRYPTO_add_lock(&ctx->references, 1, CRYPTO_LOCK_X509_STORE, "ssl.cpp", __LINE__);
 }
 
 static size_t base64_get_decoded_length(const char *data, size_t size) noexcept
