From 62c40f69adc8f2e406e5b155402326f303de47c4 Mon Sep 17 00:00:00 2001
From: Zane van Iperen <zane@zanevaniperen.com>
Date: Sun, 19 Apr 2026 04:42:16 +1000
Subject: [PATCH 2/2] util: use actual gamedir for file lookups

---
 util.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/util.cpp b/util.cpp
index ef64e4f..b81dd32 100644
--- a/util.cpp
+++ b/util.cpp
@@ -1187,7 +1187,8 @@ void UTIL_SelectWeapon(edict_t *pEdict, int weapon_index)
 
 void UTIL_BuildFileName_N(char *filename, int size, char *arg1, char *arg2)
 {
-   const char * mod_dir = (submod_id == SUBMOD_OP4) ? "gearbox" : "valve";
+   char mod_dir[256];
+   g_engfuncs.pfnGetGameDir(mod_dir);
 
    if ((arg1 != NULL) && (arg2 != NULL))
    {
-- 
2.53.0

